### PR TITLE
PMM-7482 migrate e2e tests to codecept js v3 

### PR DESCRIPTION
**Pull request by IvanaBrecherova**

### CODE REVIEW

**Comment by IvanaBrecherova** [comment]
https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests-temp/detail/pmm2-ui-tests-temp/220/tests All tests passing now.

File: pmm-app/tests/helper/grafana_helper.js [hunk 1]
```diff
@@ -12,7 +12,14 @@ class Grafana extends Helper {
     const { browserContext } = this.helpers.Playwright;
     const basicAuthEncoded = await this.getAuth(username, password);
 
-    await browserContext.setExtraHTTPHeaders({ Authorization: `Basic ${basicAuthEncoded}` });
+    for (let i = 0; i < 20; i++) {
```
Comment (puneet0191) [inline pmm-app/tests/helper/grafana_helper.js:15]: why are we doing this?

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: pmm-app/tests/helper/grafana_helper.js (hunk 1 above)
Comment (IvanaBrecherova) [reply pmm-app/tests/helper/grafana_helper.js:15]: I saw sometimes error that `browserContext` is undefined. Then a couple of tests failed on this error.

Comment (puneet0191) [reply pmm-app/tests/helper/grafana_helper.js:15]: I am not sure, please check again.

Comment (IvanaBrecherova) [reply pmm-app/tests/helper/grafana_helper.js:15]: https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests-temp/detail/pmm2-ui-tests-temp/150/tests Here is the failing. This fix solved it.

Comment (puneet0191) [reply pmm-app/tests/helper/grafana_helper.js:15]: can you please try again, it looks like a temp glitch, probably browserContext was not initialized due to browser crash, please remove and try again.

Comment (IvanaBrecherova) [reply pmm-app/tests/helper/grafana_helper.js]: I've deleted it. :)

File: pmm-app/tests/DbaaS/pages/dbaasPage.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ module.exports = {
       copyToClipboardButton: '//span[contains(text(), "Copy to clipboard")]',
       disabledAddButton: '//button[@data-qa="kubernetes-add-cluster-button" and @disabled]',
       forceUnreigsterCheckBox: '.checkbox-container__checkmark',
-      kubeconfigFileInput: '$kubeConfig-textarea-input',
+      kubeconfigFileInput: '//textarea[@data-qa=\'kubeConfig-textarea-input\']',
```
Comment (yurkovychv) [inline pmm-app/tests/DbaaS/pages/dbaasPage.js:23]:
```suggestion
      kubeconfigFileInput: '//textarea[@data-qa="kubeConfig-textarea-input"]',
```

File: pmm-app/tests/pages/adminPage.js [hunk 3]
```diff
@@ -116,7 +116,7 @@ module.exports = {
     const backgroundColor = await I.grabCssPropertyFrom(element, 'background-color');
 
     assert.strictEqual(
-      backgroundColor[0],
```
Comment (yurkovychv) [inline pmm-app/tests/pages/adminPage.js:119]: why it was removed?

File: pmm-app/tests/pages/api/securityChecksAPI.js [hunk 4]
```diff
@@ -9,8 +9,11 @@ module.exports = {
     for (let i = 0; i < timeout; i++) {
       const results = await this.getSecurityChecksResults();
 
-      if (results.length !== 0) {
-        break;
+      // waiting 10 sec to avoid error "'length' of undefined".
```
Comment (yurkovychv) [inline pmm-app/tests/pages/api/securityChecksAPI.js:12]: so where do we wait for 10 seconds?

Comment (yurkovychv) [reply pmm-app/tests/pages/api/securityChecksAPI.js:12]: I think we don't need to wait here, just an extra check for `undefined` would be enough

File: pmm-app/tests/pages/adminPage.js (hunk 3 above)
Comment (IvanaBrecherova) [reply pmm-app/tests/pages/adminPage.js:119]: The new version of codeceptjs added method `grab*FromAll` which is an array. But the `grabCssPropertyFrom` is now taking just one single element. So when you apply `backgroundColor[0]` it is taking one single letter on index 0.
 
image

File: pmm-app/tests/pages/api/securityChecksAPI.js (hunk 4 above)
Comment (IvanaBrecherova) [reply pmm-app/tests/pages/api/securityChecksAPI.js:12]: I forgot to delete this comment :) I am going to delete it.

File: pmm-app/tests/pages/adminPage.js (hunk 3 above)
Comment (yurkovychv) [reply pmm-app/tests/pages/adminPage.js:119]: ok, good

**Comment by yurkovychv** [comment]
@IvanaBrecherova CI is failing

**Comment by IvanaBrecherova** [comment]
@yurkovychv CI is running on old configuration for jenkins. This PR: https://github.com/Percona-Lab/jenkins-pipelines/pull/803 is for new configuration. I am testing it on https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests-temp/activity/

**Comment by IvanaBrecherova** [comment]
https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-tests-temp/detail/pmm2-ui-tests-temp/248/tests Tests are passing. @puneet0191 and @yurkovychv You can re-review this PR :)

File: pmm-app/tests/pages/api/securityChecksAPI.js [hunk 5]
```diff
@@ -9,8 +9,10 @@ module.exports = {
     for (let i = 0; i < timeout; i++) {
       const results = await this.getSecurityChecksResults();
 
-      if (results.length !== 0) {
-        break;
+      if (results !== undefined) {
+        if (results.length !== 0) {
+          break;
+        }
       }
```
Comment (yurkovychv) [inline pmm-app/tests/pages/api/securityChecksAPI.js:16]:
```suggestion
      if (results && results.length) {
        break;
      }
```

Comment (yurkovychv) [reply pmm-app/tests/pages/api/securityChecksAPI.js:16]: it should do the same

**Review by yurkovychv — APPROVED** [review]

**Review by puneet0191 — APPROVED** [review]

**Comment by puneet0191** [comment]
Please remember to fix pipeline when we are ready for merge to PMM-2.0

**Pull request merged**